mksnapshot.sh

```diff
@@ -9,41 +9,43 @@
 
 echo "RUN: $0"
 if [ $# -lt 1 ]; then
-	echo "Usage: $0 [branch]"
-	echo "Where [branch] is the path under /bzr/ to the branch to snapshot."
+	echo "Usage: $0 [branch] [tag]"
+	echo "Where [branch] is the name of the branch to snapshot."
+	echo "Where [tag] is the name of the tag on that branch to snapshot."
 	exit 1
 fi
 # VCS details
-module=squid3
-BZRROOT=${BZRROOT:-/bzr}
-
-# generate a tarball name from the branch ($1) note that trunk is at
-# /bzr/trunk, but we call it 3.HEAD for consistency with CVS (squid 2.x), and
-# branches are in /bzr/branches/ but we don't want 'branches/' in the tarball
-# name so we strip that.
-branchpath=${1:-trunk}
-tag=${2:-`basename $branchpath`}
+if test "x${VCSROOT}" = "x"; then
+	echo "VCSROOT needs to be set to the base path for the repository to snapshot."
+	exit 1
+fi
+
+# generate a tarball name from the branch ($1) at tag ($2)
+branch=${1:-master}
+tag=${2:-HEAD}
 startdir=${PWD}
 date=`env TZ=GMT date +%Y%m%d`
 
-tmpdir=${TMPDIR:-${PWD}}/${module}-${tag}-mksnapshot
+tmpdir=${TMPDIR:-${PWD}}/squid-${tag}-mksnapshot/
 
 rm -rf ${tmpdir}
 trap "echo FAIL-BUILD_${VERSION} ; rm -rf ${tmpdir}" 0
+mkdir ${tmpdir}
 
 rm -f ${tag}.out
-bzr export ${tmpdir} ${BZRROOT}/${module}/${branchpath} || exit 1
+git checkout ${branch} || exit 1
+(git archive --format=tar ${tag} | tar -xC ${tmpdir}) || exit 1
```

Comment:
Please check whether you need `git checkout`. IIRC, `git archive` can produce an archive without a checkout.